spring-cloud-starter-tencent-polaris-config/src/main/java/com/tencent/cloud/polaris/config/configdata/PolarisConfigDataLoader.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Tencent is pleased to support the open source community by making Spring Cloud Tencent available.
+ *
+ * Copyright (C) 2019 THL A29 Limited, a Tencent company. All rights reserved.
+ *
+ * Licensed under the BSD 3-Clause License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.tencent.cloud.polaris.config.configdata;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import com.tencent.cloud.polaris.config.adapter.PolarisConfigFilePuller;
+import com.tencent.cloud.polaris.config.adapter.PolarisPropertySourceManager;
+import com.tencent.cloud.polaris.config.config.ConfigFileGroup;
+import com.tencent.cloud.polaris.config.config.PolarisConfigProperties;
+import com.tencent.polaris.client.api.SDKContext;
+import com.tencent.polaris.configuration.api.core.ConfigFileService;
+import com.tencent.polaris.configuration.factory.ConfigFileServiceFactory;
+import org.apache.commons.logging.Log;
+
+import org.springframework.boot.ConfigurableBootstrapContext;
+import org.springframework.boot.context.config.ConfigData;
+import org.springframework.boot.context.config.ConfigDataLoader;
+import org.springframework.boot.context.config.ConfigDataLoaderContext;
+import org.springframework.boot.context.config.ConfigDataResourceNotFoundException;
+import org.springframework.boot.context.config.Profiles;
+import org.springframework.boot.logging.DeferredLogFactory;
+import org.springframework.core.env.CompositePropertySource;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.StringUtils;
+
+import static org.springframework.boot.context.config.ConfigData.Option.IGNORE_IMPORTS;
+import static org.springframework.boot.context.config.ConfigData.Option.IGNORE_PROFILES;
+import static org.springframework.boot.context.config.ConfigData.Option.PROFILE_SPECIFIC;
+
+/**
+ * Implementation of {@link ConfigDataLoader}.can be used to load {@link ConfigData} for a given
+ * {@link PolarisConfigDataResource} .
+ * <p>
+ * Load {@link ConfigData} via {@link PolarisConfigDataLoader}
+ *
+ * @author wlx
+ * @date 2022/7/5 11:14 下午
+ */
+public class PolarisConfigDataLoader implements ConfigDataLoader<PolarisConfigDataResource> {
+
+	private static final String POLARIS_CONFIG_PROPERTY_SOURCE_NAME = "polaris-config";
+
+	private final Log log;
```

Comment:
日志可以用 slf4j 的 Logger 吗？看了一下 apollo 也是用的 apache common 里的 log，是不是有特殊的原因？

Reply:
这个是在ConfigDataLoader 的javaDoc中提供的约束，我觉得可能和Spring Boot机制有关，因为Spring Boot的对slf4j的整合是完成在LoggingApplicationListener中完成的，可能在此之前的日志通过其他方式提供的
```java
/**
 * Strategy class that can be used to load {@link ConfigData} for a given
 * {@link ConfigDataResource}. Implementations should be added as {@code spring.factories}
 * entries. The following constructor parameter types are supported:
 * <ul>
 * <li>{@link Log} or {@link DeferredLogFactory} - if the loader needs deferred
 * logging</li>
 * <li>{@link ConfigurableBootstrapContext} - A bootstrap context that can be used to
 * store objects that may be expensive to create, or need to be shared
 * ({@link BootstrapContext} or {@link BootstrapRegistry} may also be used).</li>
 * </ul>
 * <p>
 * Multiple loaders cannot claim the same resource.
 *
 * @param <R> the resource type
 * @author Phillip Webb
 * @author Madhura Bhave
 * @since 2.4.0
 */
```